src/index.ts

```diff
@@ -301,6 +302,15 @@ export default async function app(
   }
 
   app.register(compatibility, { prefix: '/v1' });
+  app.register(proxy, {
+    upstream: 'https://www.google.com/s2/favicons',
+    prefix: '/icon',
+    replyOptions: {
+      queryString: (_, reqUrl) => {
+        return reqUrl.replace('/icon?url=', 'domain=').replace('size=', 'sz=');
```

Comment:
Not my favourite, but since we needed to rewrite a lot and our internal system uses this exact format it felt ok?